_episodes/24.md

```diff
@@ -0,0 +1,413 @@
+    Source: https://kubernetes.io/docs/concepts/overview/what-is-kubernetes/
+   </p>
+   
+   - Containerization simplified a trend in service architecture called 
+   microservices. Microservices deploy loosely coupled and modular applications
+   rather than all-encompassing monolithic applications. With containers, these
+   applications can be deployed and scale up quickly across various virtual and
+   physical machines without affecting applications on the same machine. This
+   was great, but new problems emerged, such as, monitoring the health of
+   applications, scaling the applications as requests grow and diminish, 
+   redeploying crashed application, and networking the applications together.
+   In summary, all of these activities can be considered container orchestration
+   and this is exactly what Kubernetes solves!
+   
+   <p align="center">
+    <img align="center" width="75%" height="100%" src="/assets/episode/24/load-balancer.jpeg"/><br/>
+    Source: https://www.slideshare.net/devopsdaysaustin/continuously-delivering-microservices-in-kubernetes-using-jenkins<br/>
+    Here we hae two services that each sit behind a load balancer provided and mapped by the Kuberenets cluster.
+   </p>
+
+1. **Kubernetes components and architecture**:
+   - Node - The physical machine or VM running a kubelet and container runtime. 
+   - Cluster - a set of nodes connected to the same control plane node.
+   - Pod - single instance of an application, the smallest object in kubernetes.
+
+   <p align="center">
+    <img align="center" width="75%" height="100%" src="/assets/episode/24/components-of-kubernetes.svg"/><br/>
+    Source: https://kubernetes.io/docs/concepts/overview/components/
+   </p>
+
+   Kubernetes control plane components:
+     - API server that kubelets connect with and is the front end for 
+     users and administrators of the cluster.
+     - etcd keystore is a distributed store containing all data used to manage 
+     the cluster
+     - Scheduler that distributes work across nodes and assigns newly created 
+     containers to nodes
+     - Controllers that are the brain behind orchestration and monitors for 
+     nodes going down etc...
+     
+   Kubernetes worker nodes components:
+     - container runtime - is the underlying runtime used to manage containers 
+     - kubelet - agent that checks the health and manages the pods running on the node based on the desired state provided in the PodSpec
+     - kube-proxy - network proxy that maintains network rules applied to nodes and allows network access between Pods in a cluster
+ 
+   You can scale up multiple PODS on a single node until the node has no more 
+   resources, at which time a new node will need to be added and pod instances will
+   be distributed between the nodes. 
+
+1. **So how does this relate to Trino?** 
+     - Why not just use docker compose? Docker-compose is a tool that creates
```

Comment:
Indent left

Reply:
also make this simpler .. or even remove docker-compose altogether ... 

If you want to keep it just use something like

- Why not just use docker-compose? Docker compose is only suitable for managing multi containers on one host. It has no orchestration features and does not support multiple hosts.